From 86862f2e9ea50fdce746688c086b2b7e0db131fb Mon Sep 17 00:00:00 2001
From: Aurelien Jarno <aurelien@aurel32.net>
Date: Thu, 30 Jun 2016 16:06:10 +0200
Subject: [PATCH] SPARC64: update localplt.data

Commits d81f90cc and 89faa0340 replaced called to __isnan and __isinf
by the corresponding GCC builtins. In turns GCC emits calls to _Qp_cmp.
We should therefore add _Qp_cmp to localplt.data as otherwise the
elf/check-localplt test fails with:

   Extra PLT reference: libc.so: _Qp_cmp

A similar change has already been done for SPARC32 in commit 6ef1cb95.

Changelog:
	* sysdeps/unix/sysv/linux/sparc/sparc64/localplt.data: Add _Qp_cmp.

(cherry picked from commit fd1cf1dc3b2d90c2a61332363feb1043f6916564)
(cherry picked from commit bbe472f4e02c18f998d8e327f4a0e7c5004c8010)
---
 sysdeps/unix/sysv/linux/sparc/sparc64/localplt.data | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysdeps/unix/sysv/linux/sparc/sparc64/localplt.data b/sysdeps/unix/sysv/linux/sparc/sparc64/localplt.data
index 633cfe4493d2..2323551f631e 100644
--- a/sysdeps/unix/sysv/linux/sparc/sparc64/localplt.data
+++ b/sysdeps/unix/sysv/linux/sparc/sparc64/localplt.data
@@ -1,4 +1,5 @@
 libc.so: _Qp_add ?
+libc.so: _Qp_cmp ?
 libc.so: _Qp_div ?
 libc.so: _Qp_feq ?
 libc.so: _Qp_fge ?
-- 
2.10.2

